File: fib.vb
Module FibModule
    Function Fib(n As Integer) As Integer
        If n <= 1 Then Return n
        Return Fib(n - 1) + Fib(n - 2)
    End Function
    Sub Main()
        Console.WriteLine(Fib(10))
    End Sub
End Module

File: fib.f
program fibmain
    implicit none
    integer :: n, result
    n = 10
    result = fib(n)
    print *, result
contains
    recursive function fib(n) result(res)
        integer, intent(in) :: n
        integer :: res
        if (n <= 1) then
            res = n
        else
            res = fib(n-1) + fib(n-2)
        end if
    end function fib
end program fibmain
